pkg/storegateway/indexcache/cache.go

```diff
@@ -65,6 +70,11 @@ type IndexCache interface {
 	// FetchSeries fetches the result of a Series() call.
 	FetchSeries(ctx context.Context, userID string, blockID ulid.ULID, matchersKey LabelMatchersKey, shard *sharding.ShardSelector) ([]byte, bool)
 
+	// StoreSeriesForPostings stores a series set for the provided postings.
+	StoreSeriesForPostings(ctx context.Context, userID string, blockID ulid.ULID, matchersKey LabelMatchersKey, shard *sharding.ShardSelector, postingsKey PostingsKey, v []byte)
```

Comment:
Why does it need all three `matchersKey`, `shard` and `postingsKey`? The first two reference the same data set as the third one, aren't they?